### ISSUE TITLE
make non-interactive mode less interactive

### ISSUE THREAD

**Issue by krohlas** [wummel/patool#55]
Hi,

First of all: Thanks for you awesome work on this API. It simplifies things a lot.

There is one issue I still need to solve, though. As you clearly state: patool does not support password encrypted archives. This is a valid design decision of the project. Nevertheless I think it should not fail because of password protected archives.

I use it in non-interactive mode like
`patoolib.extract_archive("somefile.zip", outdir="/path/to/somewhere/", interactive=False)`

and have to handle archives from an untrusted source. Some of those are password protected, I don't know which ones.

So what happens is that once I have to handle a password protected archive the non-interactive mode becomes quite interactive. Altough I don't see the output from the unpack tools used in the backend my programm stalls and waits for input. It continues once I hit enter.

So I propose to stop trying to get passwords, at least in non-interactive mode. The used tools seem to support this, but so far I did not test all of them.

I created two password protected test archives to illustrate the issue and possible solutions.

unzip fails directly when given a wrong password:

```
$ unzip -v
UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.

Latest sources and executables are at ftp://ftp.info-zip.org/pub/infozip/ ;
see ftp://ftp.info-zip.org/pub/infozip/UnZip.html for other sites.

Compiled with gcc 6.2.1 20161124 for Unix (Linux ELF).

[...]

$ unzip -P supersecret test.zip
Archive:  test.zip
   skipping: test.txt                incorrect password
```

unrar is quite interactive when being confronted with password protection:

```
$ unrar x -y test.rar

UNRAR 5.30 beta 2 freeware      Copyright (c) 1993-2015 Alexander Roshal

Enter password (will not be echoed) for test.rar:

Checksum error in the encrypted file test.rar. Corrupt file or wrong password.
No files to extract
```

We could try giving it some pseudo-password. Still interactive:

```
$ unrar x -p supersecret -y test.rar

Enter password (will not be echoed):

Reenter password:

ERROR: Passwords do not match

Enter password (will not be echoed):
[...]
```

But yeah, we can tell it to ignore password protection:

```
$ unrar x -p- -y test.rar

UNRAR 5.30 beta 2 freeware      Copyright (c) 1993-2015 Alexander Roshal

Checksum error in the encrypted file test.rar. Corrupt file or wrong password.
No files to extract
```

**Comment by krohlas** [comment]
This patch should fix it for unzip, other tools (as far as I've seen) already seem to be equipped with the needed arguments:

```
diff --git a/patoolib/programs/unzip.py b/patoolib/programs/unzip.py
index 1768f72..db7e6b3 100644
--- a/patoolib/programs/unzip.py
+++ b/patoolib/programs/unzip.py
@@ -20,6 +20,8 @@ def extract_zip (archive, compression, cmd, verbosity, interactive, outdir):
     cmdlist = [cmd]
     if verbosity > 1:
         cmdlist.append('-v')
+    if not interactive:
+        cmdlist.append('-P patoolsupersecret')
     cmdlist.extend(['--', archive, '-d', outdir])
     return cmdlist

```

**Comment by krohlas** [comment]
Another approarch: How about using https://docs.python.org/3/library/subprocess.html#subprocess.run instead of subprocess.call() for Python >=3.5 and use the input parameter here?

**Comment by yarikoptic** [comment]
What if input is requested for something else but password?

**Comment by wummel** [comment]
This is fixed in version 1.13.
